apps/meteor/client/components/GenericUpsellModal/hooks/useUpsellActions.ts

```diff
@@ -13,8 +13,8 @@ export const useUpsellActions = (hasLicenseModule = false) => {
 	const handleOpenLink = useExternalLink();
 	const cloudWorkspaceHadTrial = useSetting('Cloud_Workspace_Had_Trial', false);
 
-	const { data } = useIsEnterprise();
-	const shouldShowUpsell = !data?.isEnterprise || !hasLicenseModule;
+	const { data, isPending } = useIsEnterprise();
+	const shouldShowUpsell = !isPending && (!data?.isEnterprise || !hasLicenseModule);
```

Comment:
I would suggest using `isSuccess` instead, because it can be true with some error as well